### PR TITLE
Add transaction support

### LINKED ISSUE #29: Add transaction support
Chassis should have simple support for transactions based on https://github.com/mysqljs/mysql#transactions.

### PR DESCRIPTION
**Pull request by iamvanja**
Please let me know if there should be something fixed before the merge. 

Thanks

Closes #29

### CODE REVIEW

File: test/mysql-chassis.js [hunk 1]
```diff
@@ -316,6 +326,10 @@ describe('mysql-chassis', () => {
 
   // })
 
+  // describe('transaction methods', () => {
+
```
Comment (therealklanni) [inline test/mysql-chassis.js]: Are you planning on updating the PR with these tests later? Just want to confirm, thanks.

File: example/transaction.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import db from './connect'
+
+const insertValues = { first_name: 'Brad', last_name: 'Westfall', datetime_added: 'NOW()' }
+
+Promise.resolve()
+.then(db.beginTransaction)
```
Comment (therealklanni) [inline example/transaction.js]: If it's no trouble, could you please indent the `.then` lines so it's easier to read?

Comment (bradwestfall) [inline example/transaction.js]: I noticed that too, wasn't sure if it was just GH's rendering

Comment (iamvanja) [inline example/transaction.js]: Done.

File: test/mysql-chassis.js (hunk 1 above)
Comment (iamvanja) [inline test/mysql-chassis.js]: TBH, I have no idea how to test this without the actual interaction with the DB. So to answer your question, no plans to update this. I'd be happy to do it however if you or @bradwestfall has an idea how to do it.

Comment (therealklanni) [inline test/mysql-chassis.js]: It's pretty simple, actually. You can look at [L129](https://github.com/bradwestfall/mysql-chassis/blob/master/test/mysql-chassis.js#L129) for an example of how to instrument the tests. Basically, just tell Sinon how to resolve on each call in the tests.

Comment (iamvanja) [inline test/mysql-chassis.js]: Hi again,
I added the tests. They basically check if the chassis' methods call mysqljs' methods.
The way it works on the mysqljs' side is that all 3 methods (`beginTransaction`, `commit`, `rollback`) simply call `query` with `START TRANSACTION`, `COMMIT`, `ROLLBACK` arguments respectively. 
Is there anything else that should be tested?

**Comment by iamvanja** [comment]
Thanks for taking the time to review :)

**Comment by bradwestfall** [comment]
Thanks @iamvanja for this, I just finally had a chance to look at it for the first time, and it looks really good. Thanks @therealklanni for the wisdom as always

**Comment by bradwestfall** [comment]
Btw, just cut 4.2.0 with this

**Pull request merged**